tests/test_utility.py

```diff
@@ -31,13 +33,104 @@
                 "https://secondnode.neurobagel.org/query/": "secondnode",
             },
         ),
+        (
+            "(https://firstnode.neurobagel.org/query/, firstnode)(https://secondnode.neurobagel.org/query, secondnode)",
+            {
+                "https://firstnode.neurobagel.org/query/": "firstnode",
+                "https://secondnode.neurobagel.org/query/": "secondnode",
+            },
+        ),
+        (
+            "(https://firstnode.neurobagel.org/query/,firstnode)(https://secondnode.neurobagel.org/query,secondnode)",
+            {
+                "https://firstnode.neurobagel.org/query/": "firstnode",
+                "https://secondnode.neurobagel.org/query/": "secondnode",
+            },
+        ),
     ],
 )
 def test_parse_nodes_as_dict(monkeypatch, set_nodes, expected_nodes):
-    """Test that Neurobagel node URLs provided in a string environment variable are correctly parseed into a list."""
-    monkeypatch.setenv("NB_NODES", set_nodes)
+    """Test that Neurobagel node URLs provided in a string environment variable are correctly parsed into a list."""
+    monkeypatch.setenv("LOCAL_NB_NODES", set_nodes)
+    # TODO: This currently only compares the keys of the dicts, not the values, due to calling sorted(). This is probably not what we want.
     assert sorted(
         util.parse_nodes_as_dict(
-            os.environ.get("NB_NODES", "https://api.neurobagel.org/query/")
+            os.environ.get(
+                "LOCAL_NB_NODES", "https://api.neurobagel.org/query/"
+            )
         )
     ) == sorted(expected_nodes)
+
+
+@pytest.mark.parametrize(
+    "node_url_list, expectation, unrecognized_urls",
+    [
+        (["https://firstknownnode.org/"], does_not_raise(), None),
+        ([], does_not_raise(), None),
+        (
+            ["https://firstknownnode.org/", "https://mysterynode.org/"],
+            pytest.raises(HTTPException),
+            "['https://mysterynode.org/']",
+        ),
+        (
+            ["https://mysterynode.org/", "https://unknownnode.org/"],
+            pytest.raises(HTTPException),
+            "['https://mysterynode.org/', 'https://unknownnode.org/']",
+        ),
+    ],
+)
+def test_check_nodes_are_recognized(
+    monkeypatch, node_url_list, expectation, unrecognized_urls
+):
+    """Test that function correctly errors out when any node URL not found in the federation node index is present, but not otherwise."""
+    mock_federation_nodes = {
+        "https://firstknownnode.org/": "My First Node",
+        "https://secondknownnode.org/": "My Second Node",
+    }
```

Comment:
could you just put that directly in the `monkeypatch`? You don't seem to reuse the variable `mock_federation_nodes` again and I think it'd improve readability